src/logging/Manager.cc

```diff
@@ -1153,29 +1153,33 @@ bool Manager::WriteToFilters(const Manager::Stream* stream, zeek::RecordValPtr c
         }
 
         // Alright, can do the write now.
+        detail::LogRecord rec = RecordToLogRecord(stream, filter, columns.get());
+
+        if ( zeek::plugin_mgr->HavePluginForHook(zeek::plugin::HOOK_LOG_WRITE) ) {
+            // The current HookLogWrite API takes a threading::Value**.
+            // Fabricate the pointer array on the fly. Mutation is allowed.
+            std::vector<threading::Value*> vals(rec.size());
+            for ( size_t i = 0; i < rec.size(); i++ )
+                vals[i] = &rec[i];
```

Comment:
The pattern was to use index-assignment for the raw pointer vectors, and `emplace_back()` if a vector holds an actual `threading::Value`.

Your suggestion also requires a change to `.reserve(rec.size())` instead of the `vector(rec.size())`.

I'll change to emplace_back(), just wasn't completely random :-)